core/Plugin/Visualization.php

```diff
@@ -207,9 +207,29 @@ protected function buildView()
             $view->error = $loadingError;
         }
 
+        $templateFile = static::TEMPLATE_FILE;
+        $templateVars = $this->templateVars;
+
+        /**
+         * Can be trigerred in order to extend or modify Report visualization output.
+         *
+         * For example, if you want to display additional info which is in the dataTable but was omitted in original
+         * report visualization.
+         *
+         * Value of $templateFile should keep to Piwik format '@<Plugin_Name>/<template_name>.twig'
+         *
+         * The values under templateVars are accesible directly in template for example value under
+         * $templateVars['example'] in template is available under {{ example }}
+         *
+         * @param array &$templateFile The original template file, the overriding one should extend the original one
+         * @param array &$templateFile Template variables, you can add new or modify the ones already present
+         */
+        Piwik::postEvent('Visualization.customizeTemplate', array(&$templateFile, &$templateVars));
```

Comment:
This would make our template file names API. Instead it would be better to use Visualization ID. Also it would make the template vars API which is not ideal. Can we instead post an instance of the Visualization? Otherwise we make any template variable an API and won't be able to make any changes anymore